source/borderrouter_helpers.c

```diff
@@ -54,14 +54,11 @@ void print_memory_stats(void)
     const mem_stat_t *heap_info = ns_dyn_mem_get_mem_stat();
     if (heap_info) {
         tr_info(
-            "Heap size: %" PRIu16 ", "
-            "Reserved: %" PRIu16 ", "
-            "Reserved max: %" PRIu16 ", "
-            "Alloc fail: %" PRIu32 ""
-            ,heap_info->heap_sector_size
-            ,heap_info->heap_sector_allocated_bytes
-            ,heap_info->heap_sector_allocated_bytes_max
-            ,heap_info->heap_alloc_fail_cnt);
+            "Heap size: %d, Reserved: %d, Reserved max: %d, Alloc fail: %d"
```

Comment:
Is this because of the typedef changes? This is wrong, as you're feeding unsigned numbers to `%d` - needs to be `%u`.

If worried about portability, would be better still to use `unsigned long` and `%lu` - will deal with setups with 16-bit int but 32-bit sizes, like MSP430.